broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/steps/StoragePartitionTransitionStep.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.system.partitions.impl.steps;
+
+import io.atomix.raft.RaftServer.Role;
+import io.camunda.zeebe.broker.system.partitions.PartitionTransitionContext;
+import io.camunda.zeebe.broker.system.partitions.PartitionTransitionStep;
+import io.camunda.zeebe.util.sched.future.ActorFuture;
+import io.camunda.zeebe.util.sched.future.CompletableActorFuture;
+
+public class StoragePartitionTransitionStep implements PartitionTransitionStep {
+
+  @Override
+  public ActorFuture<Void> transitionTo(
+      final PartitionTransitionContext context, final long term, final Role role) {
+
+    switch (role) {
```

Comment:
I hope we don't have this kind of construct not in all transition steps :sweat_smile: maybe we can filter some stuff out or remap it. Might also not make sense to transition multiple times for canditate and follower

Reply:
But in general I like it that we have it in one place :+1: and see what we do in the different roles